code/game/objects/items/ornaments.dm

```diff
@@ -0,0 +1,187 @@
+/obj/item/ornament
+	name = "ornament"
+	desc = "A glass ornament. What some would call plain, others call elegant."
+	icon = 'icons/obj/ball_ornaments.dmi'
+	icon_state = "white_ball_ornament"
+	w_class = 2
+
+/obj/item/ornament/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(istype(src, /obj/item/ornament/teardrop))
```

Comment:
The pAI code does now, though I doubt pAI-integrated ornaments would ever be a thing.

But it's possible that someone else may in the future put code into `/obj/item/attackby()` or `/obj/attackby()`.